src/averageAges.js

```diff
@@ -55,7 +64,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const childrenWithMotherInList = people
+    .filter(child => people
+      .some(person => person.name === child.mother));
+  const newCildrenArray = !onlyWithSon
```

Comment:
What does `new` mean? And avoid using data type in variables names, in case if it's an array - it means that it's a list, so it's enough to use plural name or `list` in name